aptos-move/framework/aptos-token-objects/sources/token.move

```diff
@@ -77,25 +97,86 @@ module aptos_token_objects::token {
         creator_address: address,
         collection_name: String,
         description: String,
-        name: String,
+        name_prefix: String,
+        name_with_index_suffix: Option<String>,
         royalty: Option<Royalty>,
         uri: String,
     ) {
-        assert!(string::length(&name) <= MAX_TOKEN_NAME_LENGTH, error::out_of_range(ETOKEN_NAME_TOO_LONG));
+        if (option::is_some(&name_with_index_suffix)) {
+            // Be conservative, as we don't know what length the index will be, and assume worst case (20 chars in MAX_U64)
+            assert!(string::length(&name_prefix) + 20 + string::length(option::borrow(&name_with_index_suffix)) <= MAX_TOKEN_NAME_LENGTH, error::out_of_range(ETOKEN_NAME_TOO_LONG));
+        } else {
+            assert!(string::length(&name_prefix) <= MAX_TOKEN_NAME_LENGTH, error::out_of_range(ETOKEN_NAME_TOO_LONG));
+        };
         assert!(string::length(&description) <= MAX_DESCRIPTION_LENGTH, error::out_of_range(EDESCRIPTION_TOO_LONG));
         assert!(string::length(&uri) <= MAX_URI_LENGTH, error::out_of_range(EURI_TOO_LONG));
 
         let object_signer = object::generate_signer(constructor_ref);
 
         let collection_addr = collection::create_collection_address(&creator_address, &collection_name);
         let collection = object::address_to_object<Collection>(collection_addr);
-        let id = collection::increment_supply(&collection, signer::address_of(&object_signer));
+
+        // once this flag is enabled, cleanup code for aggregator_api_enabled = false.
+        let aggregator_api_enabled = features::aggregator_v2_api_enabled();
+        let concurrent_assets_enabled = aggregator_api_enabled && features::concurrent_assets_enabled();
+
+        let (deprecated_index, deprecated_name) = if (aggregator_api_enabled) {
+            let index = if (concurrent_assets_enabled) {
+                option::destroy_with_default(
+                    collection::increment_concurrent_supply(&collection, signer::address_of(&object_signer)),
+                    aggregator_v2::create_snapshot<u64>(0)
+                )
+            } else {
+                let id = collection::increment_supply(&collection, signer::address_of(&object_signer));
+                aggregator_v2::create_snapshot(option::get_with_default(&mut id, 0))
+            };
+
+            let name = if (option::is_some(&name_with_index_suffix)) {
+                aggregator_v2::string_concat(name_prefix, &index, option::extract(&mut name_with_index_suffix))
+            } else {
+                aggregator_v2::create_snapshot(name_prefix)
+            };
+
+            let deprecated_index = if (concurrent_assets_enabled) {
+                0
+            } else {
+                aggregator_v2::read_snapshot(&index)
+            };
+
+            let deprecated_name = if (concurrent_assets_enabled) {
+                string::utf8(b"")
```

Comment:
Something is strange here. So once we turn on the flags, name will be gone from all tokens? This can be quite a drastic change. What are the considerations here vs getting people to use create_indexed_token if they need indices in the name (and only setting the deprecated index & name fields to empty for that single case)?

Reply:
@movekevin Concern about that is that people will permanently need to look at both fields, and it is more likely new fields will stay broken for longer. 

This way, after people update indexer, it is going to be uniform for all. 

Cc @bowenyang007 who suggested this approach being better than the alternative